app/assets/javascripts/modules/logger.js

```diff
@@ -0,0 +1,30 @@
+(function () {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  function Logger() {
+    if (isDevelopmentEnvironment()) {
+      return console;
+    } else {
+      return silentConsole();
+    }
+  }
+
+  function isDevelopmentEnvironment() {
+    return /\.dev\.gov\.uk$/.test(window.location.hostname);
+  }
+
+  function silentConsole() {
+    var fakeConsole = {};
+
+    $.each(console, function(key) {
+      if (typeof console[key] === 'function') {
+        fakeConsole[key] = function() {};
+      }
+    });
+
+    return fakeConsole;
+  }
+
+  GOVUK.Logger = Logger;
+})();
```

Comment:
i am not convinced this is a good idea, what material difference does it make on production? and also, the way static is deployed currently, unless someone is actively working on it, this is less likely to get picked up easily.

the console is not user facing from an average users POV